digiwf-engine/digiwf-engine-service/src/main/java/io/muenchendigital/digiwf/humantask/infrastructure/repository/ActRuGroupTaskSearchRepository.java

```diff
@@ -0,0 +1,114 @@
+package io.muenchendigital.digiwf.humantask.infrastructure.repository;
+
+import io.muenchendigital.digiwf.humantask.infrastructure.entity.TaskInfoEntity;
+import io.muenchendigital.digiwf.humantask.infrastructure.entity.camunda.ActRuIdentityLinkEntity;
+import io.muenchendigital.digiwf.humantask.infrastructure.entity.camunda.ActRuTaskEntity;
+import lombok.AllArgsConstructor;
+import lombok.val;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.Nullable;
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+@AllArgsConstructor
+public class ActRuGroupTaskSearchRepository {
+    private final EntityManager em;
+    private final ActRuTaskCriteriaBuilder actRuTaskCriteriaBuilder;
+
+    public Page<ActRuTaskEntity> search(final String assigneeId, final List<String> lowerCaseGroups, final String searchQuery, final Boolean assigned, final Pageable pageable) {
```

Comment:
Nachdem der Code etwas umfangreicher ist würde ich noch einen kurzen Kommentar einfügen, der erklärt was hier passiert. Also, dass du 3 Tabellen `TaskInfoEntity`, `ActRuTaskEntity` und `actRuIdentities` zusammen joinst und dann paginiert daraus die Werte liest. Dann wird gleich auf den ersten Blick klar was die Methode macht und man muss sich nicht erstmal in die Criteria Logik reindenken

Reply:
done